presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -345,6 +354,64 @@ protected RelationType visitShowColumns(ShowColumns showColumns, AnalysisContext
     }
 
     @Override
+    protected RelationType visitShowCreateTable(ShowCreateTable showCreateTable, AnalysisContext context)
+    {
+        QualifiedObjectName tableName = createQualifiedObjectName(session, showCreateTable, showCreateTable.getName());
+
+        Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+
+        if (metadata.getView(session, tableName).isPresent()) {
+            throw new SemanticException(MISSING_TABLE, showCreateTable, "'%s' is a view, not a table", tableName);
+        }
+
+        if (!tableHandle.isPresent()) {
+            throw new SemanticException(MISSING_TABLE, showCreateTable, "Table '%s' does not exist", tableName);
+        }
+
+        ConnectorTableMetadata connectorTableMetadata = metadata.getTableMetadata(session, tableHandle.get()).getMetadata();
+
+        List<ColumnMetadata> columns = connectorTableMetadata.getColumns().stream()
+                .filter(column -> !column.isHidden())
+                .collect(toImmutableList());
+        Map<String, Object> properties = connectorTableMetadata.getProperties();
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("CREATE TABLE ");
+        builder.append(formatName(tableName.getCatalogName())).append(".");
+        builder.append(formatName(tableName.getSchemaName())).append(".");
+        builder.append(formatName(tableName.getObjectName()));
+        builder.append(" (").append(NEW_LINE);
+        String columnList = columns.stream()
+                .map(column -> "  " + formatName(column.getName()) + " " + column.getType())
+                .collect(Collectors.joining("," + NEW_LINE));
+        builder.append(columnList);
+        builder.append(NEW_LINE).append(")");
+
+        if (!properties.isEmpty()) {
+            builder.append(NEW_LINE).append("WITH (").append(NEW_LINE);
+            // Make sure we always print the map in the same way.
+            String propertyList = ImmutableSortedMap.copyOf(properties).entrySet().stream()
+                    .map(entry -> "  " + formatName(entry.getKey()) + "=" + entry.getValue())
+                    .collect(Collectors.joining("," + NEW_LINE));
+            builder.append(propertyList);
+            builder.append(NEW_LINE).append(")");
+        }
+
+        SingleColumn singleColumn = new SingleColumn(new StringLiteral(builder.toString()), Optional.of("Create Table"));
```

Comment:
Add a helper method in `QueryUtil`. Something like 'singleValueQuery(value, name)`